src/OmnichannelChatSDK.ts

```diff
@@ -2042,30 +2043,47 @@ class OmnichannelChatSDK {
                     this.scenarioMarker.completeScenario(TelemetryEvent.GetAuthToken);
                 } else {
                     const exceptionDetails = {
-                        response: "UndefinedAuthToken"
+                        response: ChatSDKErrorName.UndefinedAuthToken
                     };
 
+                    if (optionalParams?.throwError) {
+                        throw Error(exceptionDetails.response);
+                    }
+
+                    // Fail scenario only if error is not thrown then bubbled up
                     this.scenarioMarker.failScenario(TelemetryEvent.GetAuthToken, {
                         ExceptionDetails: JSON.stringify(exceptionDetails)
                     });
                 }
-            } catch {
+            } catch (error) {
                 const exceptionDetails = {
-                    response: "GetAuthTokenFailed"
+                    response: ChatSDKErrorName.GetAuthTokenFailed as string
                 };
 
+                if ((error as Error).message == ChatSDKErrorName.UndefinedAuthToken) {
+                    exceptionDetails.response = (error as Error).message;
+                }
+
                 this.scenarioMarker.failScenario(TelemetryEvent.GetAuthToken, {
                     ExceptionDetails: JSON.stringify(exceptionDetails)
                 });
+
+                if (optionalParams?.throwError) {
+                    throw Error(exceptionDetails.response);
+                }
```

Comment:
why we want to silent exceptions?,

Reply:
It's silenced during initialization for backward compatibility, but it will thrown an exception eventually in start chat if it fails.